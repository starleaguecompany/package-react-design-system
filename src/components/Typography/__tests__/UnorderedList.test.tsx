import * as React from 'react'
import { render, screen } from '@testing-library/react'

import { UnorderedList, ListItem } from '../index'

describe('UnorderedList', () => {
  test('should render correctly with no props', () => {
    const { container } = render(<UnorderedList />)

    expect(container).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('unordered', { exact: true })
  })

  test('should render correctly with ListItem', () => {
    render(
      <UnorderedList>
        <ListItem>Testing list</ListItem>
      </UnorderedList>
    )

    expect(screen.getByRole('list')).toMatchSnapshot()
    expect(screen.getByRole('list')).toContainElement(screen.getByRole('list'))
    expect(screen.getByText('Testing list')).toBeVisible()
  })

  test('should render correctly with attributes', () => {
    render(
      <UnorderedList id="test-id" className="test-class">
        Testing list
      </UnorderedList>
    )

    expect(screen.getByRole('list')).toMatchSnapshot()
    expect(screen.getByRole('list')).toHaveAttribute('id', 'test-id')
    expect(screen.getByRole('list')).toHaveClass('test-class')
  })
})
