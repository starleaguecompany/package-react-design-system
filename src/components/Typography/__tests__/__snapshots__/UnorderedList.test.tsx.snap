// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnorderedList should render correctly with ListItem 1`] = `
<ul
  class="unordered"
>
  <li
    class="item"
  >
    Testing list
  </li>
</ul>
`;

exports[`UnorderedList should render correctly with attributes 1`] = `
<ul
  class="test-class unordered"
  id="test-id"
>
  Testing list
</ul>
`;

exports[`UnorderedList should render correctly with no props 1`] = `
<div>
  <ul
    class="unordered"
  />
</div>
`;
